File: src/main/java/online/maestoso/soulforged/sound/SoulforgedSoundEvents.java
package online.maestoso.soulforged.sound;

import net.minecraft.sound.SoundEvent;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class SoulforgedSoundEvents {
    public static final SoundEvent CRIT_THRUSTING = register("soulforged:crit_thrusting");
    public static final SoundEvent CRIT_SLASHING = register("soulforged:crit_slashing");
    public static final SoundEvent CRIT_CRUSHING = register("soulforged:crit_crushing");

    private static SoundEvent register(String identifier) {
        Identifier id = Identifier.tryParse(identifier);
        SoundEvent se = new SoundEvent(id);
        Registry.register(Registry.SOUND_EVENT, id, se);
        return se;
    }
}

File: src/main/kotlin/studio/soulforged/soulforged/sound/SoulforgedSoundEvents.kt
package studio.soulforged.soulforged.sound

import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.sound.SoundEvent
import net.minecraft.util.Identifier
import studio.soulforged.soulforged.Soulforged.sid

@Suppress("unused")
object SoulforgedSoundEvents {
    val CRIT_THRUSTING = register("crit_thrusting".sid())
    val CRIT_SLASHING = register("crit_slashing".sid())
    val CRIT_CRUSHING = register("crit_crushing".sid())

    private fun register(identifier: String): SoundEvent {
        val id = Identifier.tryParse(identifier)
        val se = SoundEvent.createVariableRangeEvent(id)
        Registry.register(Registries.SOUND_EVENT, id, se)
        return se
    }
    internal fun init() {

    }
}